src/simulation/m_riemann_solvers.fpp

```diff
@@ -1764,20 +1785,28 @@ contains
                                     end do
                                 end if
 
+                                if (low_Mach == 2) then
+                                    zcoef = min(1d0, max(vel_L_rms**0.5d0/c_L, vel_R_rms**0.5d0/c_R))
```

Comment:
these `if(low_mach = 1` (or `2`) routines seem to be repeated for all of the types of Riemann solver. they should be turned into their own functions or subroutines.